src/signature.rs

```diff
@@ -30,33 +30,35 @@ impl Signature {
                     algorithm: "http://www.w3.org/2001/10/xml-exc-c14n#".to_string(),
                 },
                 signature_method: SignatureMethod {
-                    algorithm: "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256".to_string(),
+                    algorithm: SignatureAlgorithm::RsaSha256,
```

Comment:
People using this library are likely using more than just these values. If you want to make this an enum, we need an override to set a custom string here

Reply:
Maybe the name `Unsupported` was poorly chosen, but isn't that case covered by the `SignatureAlgorithm::Unsupported(String)` variant?